### PR TITLE
Migrate release/5.0 to netcoreapp3.1

### PR DESCRIPTION
**Pull request by AlitzelMendez**
In order to fix governance compliance 

Port https://github.com/dotnet/arcade/pull/6891
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by riarenas** [comment]
Is this related to component governance? removing netcoreapp2.1 support in a servicing branch is a bit scary. I'm guessing we can't just suppress the EOL warnings from the sdk?

File: eng/common/templates/post-build/post-build.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ parameters:
   PromoteToChannelIds: ''
 
   enableSourceLinkValidation: false
-  enableSigningValidation: true
```
Comment (riarenas) [inline eng/common/templates/post-build/post-build.yml:19]: I think I know why you're turning this off, but it should have either something about this in the PR description or some comment here to mention it's only temporary while you produce an updated sdk. (maybe even both things)

**Comment by AlitzelMendez** [comment]
> Which parts of this are related to component governance? removing netcoreapp2.1 support in a servicing branch is a bit scary. I'm guessing we can't just suppress the EOL warnings from the sdk?

oh if that is a plausible option it sounds good for me, I just would like to double check with @garath as he has been driving a lot of stuff of similar efforts and we talked about this last week :)

File: eng/common/templates/post-build/post-build.yml (hunk 1 above)
Comment (mmitche) [reply eng/common/templates/post-build/post-build.yml:19]: Yeah that would be good

**Comment by mmitche** [comment]
@AlitzelMendez How much of this port was conflict-free? It looks pretty straightforward, just wondering how much we may or may not have missed.

**Comment by garath** [comment]
> I just would like to double check with @garath

I defer to @mmitche. I do not know the state of 2.1 enough to make this call.

**Comment by mmitche** [comment]
> > I just would like to double check with @garath
> 
> I defer to @mmitche. I do not know the state of 2.1 enough to make this call.

I'm fine with just suppressing the warnings. 2.1 is definitely out of public support, but so is 5.0, and netcoreapp3.1 will be at the end of the year too.

So rolling this forward to 3.1 doesn't help us too much

**Pull request closed without merging**